holmes/plugins/toolsets/grafana/common.py

```diff
@@ -19,20 +22,96 @@ def headers(api_key: str):
     }
 
 
+def is_int(string):
+    try:
+        int(string)
+    except ValueError:
+        return False
+    else:
+        return True
+
+
+def is_rfc3339(timestamp_str: str) -> bool:
+    """Check if a string is in RFC3339 format."""
+    try:
+        parser.parse(timestamp_str)
+        return True
+    except (ValueError, TypeError):
+        return False
+
+
+def rfc3339_to_unix(timestamp_str: str) -> int:
+    dt = parser.parse(timestamp_str)
+    return int(dt.timestamp())
+
+
+def unix_to_rfc3339(timestamp: int) -> str:
+    dt = datetime.datetime.fromtimestamp(timestamp, datetime.timezone.utc)
+    return dt.isoformat()
+
+
 def process_timestamps(
     start_timestamp: Optional[Union[int, str]], end_timestamp: Optional[Union[int, str]]
-):
-    if start_timestamp and isinstance(start_timestamp, str):
-        start_timestamp = int(start_timestamp)
-    if end_timestamp and isinstance(end_timestamp, str):
-        end_timestamp = int(end_timestamp)
+) -> Tuple[str, str]:
+    """
+    Process and normalize start and end timestamps.
 
+    Supports:
+    - Integer timestamps (Unix time)
+    - RFC3339 formatted timestamps
+    - Negative integers as relative time from the other timestamp
+    - Auto-inversion if start is after end
+
+    Returns:
+        Tuple of (start_timestamp, end_timestamp)
+    """
+    # If no end_timestamp provided, use current time
     if not end_timestamp:
         end_timestamp = int(time.time())
+
+    # If no start_timestamp provided, default to one hour before end
     if not start_timestamp:
-        start_timestamp = end_timestamp - ONE_HOUR_IN_SECONDS
-    if start_timestamp < 0:
-        start_timestamp = end_timestamp + start_timestamp
+        start_timestamp = -ONE_HOUR_IN_SECONDS
+
+    if start_timestamp and is_int(start_timestamp):
```

Comment:
Please refactor the whole if else to a function and reuse.
def try_parse_datetime()